eospac: add version 6.4.2beta

gaber opened the pull request.

gaber committed:
eospac: add version 6.4.2beta

adamjstewart commented on var/spack/repos/builtin/packages/eospac/package.py:
```diff
@@ -15,7 +15,9 @@ class Eospac(Package):
     list_url = "http://laws.lanl.gov/projects/data/eos/eospacReleases.php"
 
-    version('6.4.1',       sha256='2310c49bd7a60cad41d2cb1059c5f0a1904f0c778b164937182382df326ca003', preferred=True,
+    version('6.4.2beta',   sha256='635b94f1ec7558deca92a3858c92db0f4437170252bb114cbdb809b74b6ee870', preferred=True,
```
We don't want to install a beta version by default, you should keep `preferred=True` on the last stable release.

gaber replied:
It's unfortunate that EOSPAC's maintainers use the label "beta" to describe some versions, but the difference between 6.4.1 and 6.4.2beta is only one bug fix, for a bug we reported. I'm happy to move `preferred=True` back to the last "stable" release, but no one should actually prefer that previous version.

adamjstewart replied:
In that case, you can make your new version preferred. It might be useful to add a comment to the package saying that beta doesn't mean beta.

gaber replied:
Sounds good, thanks. I've added a comment above the list of versions; let me know if that's still too muddy.

gaber committed:
eospac: clarify EOSPAC "beta" versions

Compared to 6.4.1, EOSPAC 6.4.2beta contains only one change, a fix
for an inability to read some SESAME files in ASCII format.  From the
release announcement,

    EOSPAC 6.4.2beta has been released for general use as the latest
    (i.e., eospac6-latest) versions.  This is a small patch to the
    previously-released version 6.4.1, which was requested by an
    affected user.

But the "beta" label can cause confusion, especially when a beta
version is the new preferred version, as is the case here.  As
suggested by reviewers, add a comment clarifying EOSPAC's use of
"beta".
